### ISSUE TITLE
Error comiling

### ISSUE THREAD

**Issue by nitingupta910** [rusterlium/NifIo#1]
```
NifIo$ mix compile
Compiling NIF crate '/home/ngupta/temp/NifIo/native/io' (release)...
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading rustler v0.9.1
 Downloading bufstream v0.1.2
 Downloading rustler_codegen v0.9.1
 Downloading erlang_nif-sys v0.6.1
 Downloading easy-plugin v0.7.3
 Downloading aster v0.21.1
 Downloading easy-plugin-plugins v0.7.3
   Compiling erlang_nif-sys v0.6.1
   Compiling lazy_static v0.1.16
   Compiling bufstream v0.1.2
   Compiling void v1.0.2
   Compiling easy-plugin-plugins v0.7.3
   Compiling aster v0.21.1
   Compiling unreachable v0.1.1
   Compiling rustler v0.9.1
/home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/rustler-0.9.1/src/wrapper/nif_interface.rs:126:68: 126:72 error: mismatched types [E0308]
/home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/rustler-0.9.1/src/wrapper/nif_interface.rs:126     erlang_nif_sys::enif_open_resource_type(env, module_str, name, dtor, flags, tried)
                                                                                                                                                                               ^~~~
/home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/rustler-0.9.1/src/wrapper/nif_interface.rs:126:68: 126:72 help: run `rustc --explain E0308` to see a detailed explanation
/home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/rustler-0.9.1/src/wrapper/nif_interface.rs:126:68: 126:72 note: expected type `std::option::Option<unsafe extern "C" fn(*mut types::erlang_nif_sys::ErlNifEnv, *mut std::os::raw::c_void)>` 
/home/ngupta/.cargo/registry/src/github.com-1ecc6299db9ec823/rustler-0.9.1/src/wrapper/nif_interface.rs:126:68: 126:72 note:    found type `std::option::Option<extern "C" fn(*mut types::erlang_nif_sys::ErlNifEnv, *mut std::os::raw::c_void)>` 
error: aborting due to previous error 
Build failed, waiting for other jobs to finish...
error: Could not compile `rustler`.

To learn more, run the command again with --verbose.
** (RuntimeError) Compile error (exit code 101)
    lib/compiler.ex:61: Mix.Tasks.Compile.Rustler.compile_crate/2
    (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
    (mix) lib/mix/task.ex:294: Mix.Task.run_task/3
    (elixir) lib/enum.ex:1229: Enum."-map/2-lists^map/1-0-"/2
    (mix) lib/mix/tasks/compile.all.ex:19: anonymous fn/1 in Mix.Tasks.Compile.All.run/1
    (mix) lib/mix/tasks/compile.all.ex:37: Mix.Tasks.Compile.All.with_logger_app/1
    (mix) lib/mix/task.ex:294: Mix.Task.run_task/3
    (mix) lib/mix/tasks/compile.ex:85: Mix.Tasks.Compile.run/1

```

```
~/temp/NifIo$ rustc --version
rustc 1.17.0-nightly (0aeb9c129 2017-03-15)
```

**Comment by scrogson** [comment]
This should be fixed now with the merging of #3.

**Comment by philss** [comment]
I'm assuming it works now. Closing.
